src/OrchardCore/OrchardCore.ContentManagement/DefaultContentManager.cs

```diff
@@ -242,6 +242,22 @@ public async Task<ContentItem> GetVersionAsync(string contentItemVersionId)
             return await LoadAsync(contentItem);
         }
 
+        public async Task SaveDraftAsync(ContentItem contentItem)
+        {
+            if (!contentItem.Latest || contentItem.Published)
```

Comment:
Hi @jtkech - this line doesn't seem right as it prevents handlers being fired when adding a draft to an already published content item.

Reply:
@sturatcliffe Thanks for the feedback

- Normally only one content item version can be `Published` and only version can be the `Latest`.

- An active version is `Published` OR the `Latest` (so can be both). So a version is not active if it is neither `Published` nor the `Latest`, there is a pending audit trail module to manage these archived versions, otherwise we don't take them into account because they are considered as soft deleted versions.

- So let's only talk about active versions, if a version is `Published` and `Latest` it means that there is no active draft version for the related content item. When we publish it again, a new version is created that is marked as `Published` and `Latest`, and we set to false the `Published` and `Latest` of the previous version that becomes an archived version.

- If we save as a draft this `Published` and `Latest`, a new version is created that is only marked as `Latest`, the published version is still the same but no more the `Latest` and we now have an active draft version. When we edit a content item we always edit the `Latest` version, so here it will be the new active draft version. If you save it again as a draft version, we don't create a new version, we just update this version that is still the active draft version.

Sorry for the long story and there are different ways to describe content item versions ;)

**So what i can say here is that an active draft version is always the `Latest` and of course can't be `Published`**.

And `SaveDraftAsync()` is intended to call the related handlers only if we are saving a new active draft version.

I don't know in which context you are using `SaveDraftAsync()` but at this point your content item version needs to be the `Latest` and you have to take care that you have only one `Latest`. For this, e.g. when updating a content item, we first get this content item through the content manager by using the option `DraftRequired` this will create a non published version but that becomes the latest.